### PR TITLE
Validate App CR cm/secret if managed by config-controller

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/14604
Towards https://github.com/giantswarm/giantswarm/issues/14603

### CODE REVIEW

**Review by rossf7 — APPROVED** [review]
@corest Thanks LGTM

But can you add a case to the UT here?

https://github.com/giantswarm/app/blob/ac640c753661bda99ed1401768c2afc505d898c5/pkg/validation/app_test.go

File: pkg/validation/app.go [hunk 1]
```diff
@@ -64,6 +65,12 @@ func (v *Validator) validateCatalog(ctx context.Context, cr v1alpha1.App) error
 }
 
 func (v *Validator) validateConfig(ctx context.Context, cr v1alpha1.App) error {
+	_, hasManagedConfig := cr.Annotations[annotation.ConfigMajorVersion]
+	if hasManagedConfig && (key.AppConfigMapName(cr) == "" || key.AppSecretName(cr) == "") {
+		// wait for config-controller setting app CR configmap and secret
+		return microerror.Mask(appDependencyNotReadyError)
```
Comment (kopiczko) [inline pkg/validation/app.go:71]:
```suggestion
		return microerror.Maskf(appDependencyNotReadyError, "user ConfigMap or Secret not set")
```

Comment (corest) [reply pkg/validation/app.go:71]: But it is not `user` configmap or secret. I'll set then `ConfigMap or Secret not set`

Comment (kopiczko) [reply pkg/validation/app.go]: Edited.

**Comment by corest** [comment]
@kopiczko I've updated the config version here

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Validate App CR configmap and secret names if managed by config-controller.
+
+### Changed
+
+- Change `config.giantswarm.io/major-version` annotation to `config.giantswarm.io/version`.
```
Comment (kopiczko) [inline CHANGELOG.md:16]:
```suggestion
- Change (unused yet) `config.giantswarm.io/major-version` annotation to `config.giantswarm.io/version`.
```

**Review by kopiczko — APPROVED** [review]
Let's go, I need this :)

**Pull request merged**